### ISSUE TITLE
[NV-1557] 🚀 Feature: Allow pagination handling in refetch method

### ISSUE THREAD

**Issue by JoaoMagnago** [novuhq/novu#2561]
### 🔖 Feature description

A way of selecting a number of pages to refetch in novu-notification-center's `refetch` method.

### 🎤 Why is this feature needed ?

I would like to be able to refetch the user's paginated notifications on the specific page they were prior to the refetch action.

Example: When a user has multiple unseen notifications and triggers a `markAsSeen` method in a notification situated in a page different than 0 in the pagination, a `refetch` method is called to update the notifications array and confirm to the user that the notification was indeed marked as seen. But the refetch method resets the page to 0 and, to view that single notification, they would have to load possibly multiple pages. If there was a way to inform the refetch method of how many pages to load at once, the user would likely not even notice the reload, and would be able to keep viewing the notifications from that point on.

### ✌️ How do you aim to achieve this?

I would like to be able to pass a number of pages to refetch as a parameter in the `refetch` method.

Implementation (using Typescript):
```typescript
function refetch({ page: number }) {
  // Refetches on desired page
}
```

Usage (using React and Typescript):
```jsx

const [currentPage, setCurrentPage] = useState(0);

function handleLoadMore() {
  if(hasNextPage) {
    fetchNextPage();
    setCurrentPage(currentPage + 1);
  }
}

function handleMarkAsSeen(notificationId: string) {
  markAsSeen(notificationId);
  refetch({ page: currentPage });
}
```

### 🔄️ Additional Information

I tried manually fetching the pages after the refetch, but calling the fetchNextPage method multiple times in a short period of time caused some wrong behaviors, such as not loading all the desired pages and missing notifications. And even if those behaviors did not happen, calling a fetch method to an API multiple times in a row isn't performatic at all.

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

None

<sub>[NV-1557](https://linear.app/novu/issue/NV-1557/🚀-feature-allow-pagination-handling-in-refetch-method)</sub>

**Comment by scopsy** [comment]
Thank you for the report! Behind the scenes we are using react-query refetch mechanism for the whole paginated queries. I think the main problem you are describing is that the page is reseted to 0 instead of staying in the current page. Will add a help wanted label for this

**Comment by JoaoMagnago** [comment]
> Thank you for the report! Behind the scenes we are using react-query refetch mechanism for the whole paginated queries. I think the main problem you are describing is that the page is reseted to 0 instead of staying in the current page. Will add a help wanted label for this

Thank you for the update! Yes, the main problem is that the page is reset to 0 instead of staying in the current page. Looking forward to it. If I have any ideas as well I will be happy to contribute.

**Comment by gitstart** [comment]
@scopsy I would like to pick this up

**Comment by LetItRock** [comment]
hi @JoaoMagnago!
I assume that you want to refetch notifications because you've marked one or many notifications as seen right?
If that's true, the call to the `markNotificationAsSeen` or `markAllNotificationsAsSeen` should be enough to cover your case you don't have to call `refetch` method because these functions on the success will update the notifications in the array for you automatically. These functions should be available from `v0.11.0`.

**Comment by JoaoMagnago** [comment]
> hi @JoaoMagnago! I assume that you want to refetch notifications because you've marked one or many notifications as seen right? If that's true, the call to the `markNotificationAsSeen` or `markAllNotificationsAsSeen` should be enough to cover your case you don't have to call `refetch` method because these functions on the success will update the notifications in the array for you automatically. These functions should be available from `v0.11.0`.

That's great! This could solve my problem. But the pagination handling in refetch might still be useful for other purposes as well, so it will be useful to still follow up with the current ongoing PR.

Thank you. Cheers.

**Comment by LetItRock** [comment]
> > hi @JoaoMagnago! I assume that you want to refetch notifications because you've marked one or many notifications as seen right? If that's true, the call to the `markNotificationAsSeen` or `markAllNotificationsAsSeen` should be enough to cover your case you don't have to call `refetch` method because these functions on the success will update the notifications in the array for you automatically. These functions should be available from `v0.11.0`.
> 
> That's great! This could solve my problem. But the pagination handling in refetch might still be useful for other purposes as well, so it will be useful to still follow up with the current ongoing PR.
> 
> Thank you. Cheers.

I'm not sure about it, refetching on a particular page might be a bad decision if the data changes often. And because we do use query-cache then the pages data might be in the wrong state, compared to the state on the server.
